intel-types/src/lib.rs

```diff
@@ -52,12 +54,267 @@ impl<T: Copy + Debug + PartialEq + BitAnd<Output = T>> PartialEq<T> for Masked<T
+        const OVERFLOW = 1 << 3;
+
+        /// Underflow Flag
+        const UNDERFLOW = 1 << 4;
+
+        /// Precision Flag
+        const PREC = 1 << 5;
+
+        /// Denormals are Zeros
+        const DENORM_ARE_ZEROS = 1 << 6;
+
+        /// Invalid Operation Mask
+        const INV_OP_MASK = 1 << 7;
+
+        /// Denormal Operation Mask
+        const DENORM_MASK = 1 << 8;
+
+        /// Divide by Zero Mask
+        const ZERO_DIV_MASK = 1 << 9;
+
+        /// Overflow Mask
+        const OVERFLOW_MASK = 1 << 10;
+
+        /// Underflow Mask
+        const UNDERFLOW_MASK = 1 << 11;
+
+        /// Precision Mask
+        const PREC_MASK = 1 << 12;
+
+        /// Rounding Control 0
+        const ROUND_CTRL0 = 1 << 13;
+
+        /// Rounding Control 1
+        const ROUND_CTRL1 = 1 << 14;
+
+        /// Flush to Zero
+        const FLUSH_TO_ZERO = 1 << 15;
+    }
+}
+
+bitflags! {
+    /// Section 13.4.3
+    pub struct XcompBv: u64 {
+
+        /// Compacted form is used for the layout of the XSAVE EXtended Region
+        const COMPACT = 1 << 63;
+    }
+}
+
+/// The legacy region of an XSAVE area comprises the 512 bytes starting at the area's base address.
+/// See Table 13-1.
+#[repr(C)]
+pub struct XSaveLegacy {
+    /// x87 Floating Point Unit (FPU) Control Word
+    pub fcw: Fcw,
+
+    /// x87 FPU Status Word
+    pub fsw: u16,
+
+    /// x87 FPU Tag Word
+    pub ftw: u8,
+
+    /// Reserved
+    pub reserved0: u8,
+
+    /// x87 FPU Opcode
+    pub fop: u16,
+
+    /// x87 FPU Instruction Pointer Offset
+    pub fip: u64,
+
+    /// x87 FPU Data Pointer Offset
+    pub fdp: u64,
+
+    /// 32-bit register providing status and control bits used in SIMD floating-point operations
+    pub mxcsr: Masked<MxCsr>,
+
+    /// Register used to perform operations on 64-bit packed integer data
+    pub mm: [Mm; 8],
+
+    /// 128-bit data register used to operate on packed single-precision floating-point operands
+    pub xmm: [Xmm; 16],
+
+    /// Padding: Size of XSaveLegacy must be 512.
+    pub padding0: [u64; 11],
+
+    /// Padding: Size of XSaveLegacy must be 512.
+    pub padding1: [u8; 7],
+}
+
+impl Default for XSaveLegacy {
+    /// The x87 state initial configuration for FCW masks all floating-point exceptions,
+    /// sets rounding to nearest, and sets the x87 FPU precision to 64 bits.
+    /// See Section 8.1.5.
+    ///
+    /// The initial state of MXCSR after power-up/reset or INIT; mask allows software
+    /// to identify any reserved bits in MXCSR (none are reserved here).
+    /// See Sections 11.6.4 and 11.6.6.
+    fn default() -> Self {
+        let mut xsl = MaybeUninit::<XSaveLegacy>::zeroed();
+        unsafe {
+            (*xsl.as_mut_ptr()).fcw = Fcw::from_bits(0x037F).unwrap();
+        }
+        unsafe {
+            (*xsl.as_mut_ptr()).mxcsr = Masked::<MxCsr> {
+                data: MxCsr::from_bits(0x1F80).unwrap(),
+                mask: MxCsr::from_bits(0xFFF).unwrap(),
+            }
+        }
```

Comment:
This unsafe code isn't needed here. Figure out what the raw values mean in terms of flags and set those.

Reply:
I might need some guidance here; did you want me to set the entire `XSaveLegacy` region as `bitflags!`? Other than this `unsafe` hack, I'm reading that it's not possible to create a `MaybeUninit` struct and then write to its fields: https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initializing-a-struct-field-by-field

Reply:
I've updated the flags; though as mentioned I don't know how to get rid of the `unsafe` block.